pkg/storage/unified/resource/storage_backend_test.go

```diff
@@ -0,0 +1,1030 @@
+	require.Equal(t, int32(404), response.Error.Code)
+	require.Equal(t, "not found", response.Error.Message)
+
+	// Try to read the original version (should still work)
+	readReq.ResourceVersion = rv1
+	response = backend.ReadResource(ctx, readReq)
+	require.Nil(t, response.Error, "ReadResource should succeed for specific version before deletion")
+	require.Equal(t, rv1, response.ResourceVersion)
+	require.Equal(t, objectToJSONBytes(t, testObj), response.Value)
+}
+
+func TestKvStorageBackend_ListIterator_Success(t *testing.T) {
+	backend := setupTestStorageBackend(t)
+	ctx := context.Background()
+
+	// Create multiple test resources
+	resources := []struct {
+		name  string
+		group string
+		value string
+	}{
+		{"resource-1", "apps", "data-1"},
+		{"resource-2", "apps", "data-2"},
+		{"resource-3", "core", "data-3"},
+	}
+
+	for _, res := range resources {
+		testObj, err := createTestObjectWithName(res.name, res.group, res.value)
+		require.NoError(t, err)
+
+		metaAccessor, err := utils.MetaAccessor(testObj)
+		require.NoError(t, err)
+
+		writeEvent := WriteEvent{
+			Type: resourcepb.WatchEvent_ADDED,
+			Key: &resourcepb.ResourceKey{
+				Namespace: "default",
+				Group:     res.group,
+				Resource:  "deployments",
+				Name:      res.name,
+			},
+			Value:      objectToJSONBytes(t, testObj),
+			Object:     metaAccessor,
+			PreviousRV: 0,
+		}
+
+		_, err = backend.WriteEvent(ctx, writeEvent)
+		require.NoError(t, err)
+	}
+
+	// Test listing all resources in "apps" group
+	listReq := &resourcepb.ListRequest{
+		Options: &resourcepb.ListOptions{
+			Key: &resourcepb.ResourceKey{
+				Namespace: "default",
+				Group:     "apps",
+				Resource:  "deployments",
+			},
+		},
+		Limit: 10,
+	}
+
+	var collectedItems []struct {
+		name            string
+		namespace       string
+		resourceVersion int64
+		value           []byte
+	}
+
+	rv, err := backend.ListIterator(ctx, listReq, func(iter ListIterator) error {
+		for iter.Next() {
+			if err := iter.Error(); err != nil {
+				return err
+			}
+			collectedItems = append(collectedItems, struct {
+				name            string
+				namespace       string
+				resourceVersion int64
+				value           []byte
+			}{
+				name:            iter.Name(),
+				namespace:       iter.Namespace(),
+				resourceVersion: iter.ResourceVersion(),
+				value:           iter.Value(),
+			})
+		}
+		return iter.Error()
+	})
+
+	require.NoError(t, err)
+	require.Greater(t, rv, int64(0))
+	require.Len(t, collectedItems, 2) // Only resources in "apps" group
+
+	// Verify the items contain expected data
+	names := make([]string, len(collectedItems))
+	for i, item := range collectedItems {
+		names[i] = item.name
+		require.Equal(t, "default", item.namespace)
+		require.Greater(t, item.resourceVersion, int64(0))
+		require.NotEmpty(t, item.value)
+	}
+	require.Contains(t, names, "resource-1")
+	require.Contains(t, names, "resource-2")
```

Comment:
```suggestion
	require.Equal(t, []string{"resource-1", "resource-2"}, names)
```

This also checks ordering. WDYT?